modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -401,6 +401,47 @@ void ONNXImporter::populateNet(Net dstNet)
             layerParams.set("pool", layer_type == "GlobalAveragePool" ? "AVE" : "MAX");
             layerParams.set("global_pooling", true);
         }
+        else if (layer_type == "Slice")
+        {
+            if (layerParams.has("steps")) {
+                DictValue steps = layerParams.get("steps");
+                for (int i = 0; i < steps.size(); ++i) {
+                    if (steps.get<int>(i) != 1)
+                        CV_Error(Error::StsNotImplemented,
+                                 "Slice layer only supports steps = 1");
+                }
+            }
+
+            if (layerParams.has("axes")) {
+                DictValue axes = layerParams.get("axes");
+                for (int i = 1; i < axes.size(); ++i) {
+                    CV_Assert(axes.get<int>(i - 1) == axes.get<int>(i) - 1);
+                }
+                layerParams.set("axis", axes.get<int>(0));
+            } else
+                layerParams.set("axis", 0);
+
+            int axis = layerParams.get<int>("axis");
+            if (axis > 0)
+            {
+                DictValue starts = layerParams.get("starts");
+                DictValue ends = layerParams.get("ends");
+                CV_Assert(starts.size() == ends.size());
+                std::vector<int> begin(axis, 0);
+                std::vector<int> end(axis, -1);
+
+                for (int i = 0; i < starts.size(); ++i) {
+                    begin.push_back(starts.get<int>(i));
+                    int finish = ends.get<int>(i);
+                    end.push_back((finish < 0) ? --finish : finish);
+                }
+                layerParams.set("begin", DictValue::arrayInt(&begin[0], begin.size()));
+                layerParams.set("end", DictValue::arrayInt(&end[0], end.size()));
```

Comment:
If we put just `begins`, `ends` and `axis > 0` can we do the logic with `Range::all` not only for dimensions after specified in the parameters but also for dimensions before `axis`?